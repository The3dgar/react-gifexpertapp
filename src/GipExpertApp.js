import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GipExpertApp = ({defaultCategories= []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <h2>GipExpertApp</h2>
      <AddCategory setCategories={setCategories}></AddCategory>
      <ol>
        {categories.map((c) => (
          <GifGrid key={c} category={c}></GifGrid>
        ))}
      </ol>
    </div>
  );
};

export default GipExpertApp;
