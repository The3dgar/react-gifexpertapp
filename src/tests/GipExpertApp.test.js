import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GipExpertApp from "../GipExpertApp";

describe("pruebas en <GipExpertApp></GipExpertApp>", () => {
  test("should render GifGridItem", () => {
    const wrapper = shallow(<GipExpertApp></GipExpertApp>);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render una lista de categorias", () => {
    const categories = ["One Punch", "Friends"];

    const wrapper = shallow(
      <GipExpertApp defaultCategories={categories}></GipExpertApp>
    );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find("GifGrid").length).toBe(categories.length)

  });
});
