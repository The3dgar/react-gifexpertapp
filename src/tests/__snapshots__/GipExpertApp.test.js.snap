// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GipExpertApp></GipExpertApp> should render GifGridItem 1`] = `
<div>
  <h2>
    GipExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</div>
`;

exports[`pruebas en <GipExpertApp></GipExpertApp> should render una lista de categorias 1`] = `
<div>
  <h2>
    GipExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Friends"
      key="Friends"
    />
  </ol>
</div>
`;
